### PR TITLE
Migrate the Worktree API to Protobuf

### PR DESCRIPTION
**Pull request by Nek-12**
## Summary

- Migrates the complete Worktree API vertical from handwritten JSON contracts to generated binary Protobuf contracts across the server, Go clients, CLI/TUI, and Desktop.
- Uses generated descriptors as operation authority and generated messages/validation as wire authority, while removing the legacy Worktree DTO, route, schema, and numeric-error implementations.
- Preserves current Worktree behavior, including synchronous Delete and scheduled Enter/Leave setup outcomes, and advances the protocol version from 140 to 141.

## QA and verification

- The implementation completed generation, server/Desktop lint dry-runs, Go/Desktop builds, and `git diff --check` during review.
- After the final scope correction, `just gen`, `just build go`, and `git diff --check` passed.
- PR review remediation additionally passed the focused existing `./shared/client` tests and `just build go`.
- No final manual QA pass was run in the review-fix round. QA identified the existing linked-Workspace Git-primary deletion edge case; the User confirmed it is valid but outside this behavior-preserving migration and it is tracked in KENT-615.

## Diff size

Against the integrated `origin/main` ancestor (`ea9579c5a`):

- Production code: **+4,168 / -4,185** lines; 8,353 gross changed lines; net **-17**.
- Test/generated/docs code: **+2,345 / -2,985** lines; 5,330 gross changed lines; net **-640**. This category contains existing test/test-support ports; tracked generated output and documentation have no diff.
- Total: **+6,513 / -7,170** lines; 13,683 gross changed lines; net **-657**.

## Caveats and follow-up

- This is an intentional hard cutover: prior clients using the Worktree JSON contract are rejected by the protocol-version boundary; there is no fallback or dual transport.
- Canonical generated Protobuf JSON changes the public `kent worktree ... --json` shape as explicitly approved for this migration.
- Generated Go/TypeScript/descriptors are ignored build outputs and must be regenerated from the checked-in schemas.
- KENT-615 tracks blocking deletion of Git’s primary worktree when a Kent Workspace is rooted at a linked worktree. That behavior change was intentionally excluded here to preserve existing Worktree semantics.

## Tracking

- Kent task: KENT-557
- Follow-up: KENT-615
- GitHub issue: none.

### CODE REVIEW

File: shared/client/remote_binary_subscription.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		return nil, err
+	}
+	conn, cleanup, err := c.openRPCConn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	startResult := &worktreepb.SetupStartResult{}
+	if err := callBinaryRPC(ctx, conn, operations.Subscribe.Name, method, request, startResult); err != nil {
+		cleanup()
+		return nil, errors.Join(serverapi.ErrStreamFailed, err)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply shared/client/remote_binary_subscription.go:35]: Fixed in fa58e6659. `binaryCallFrame` is the shared client-to-server binary call encoder, and the unary-only kind check incorrectly prevented the generated subscription start from using it. Removing that check lets `SubscribeWorktreeSetup` send and receive its generated start Result on the dedicated connection; unary connection-policy enforcement remains at the unary dispatch owners. Focused existing `./shared/client` tests and `just build go` pass.

File: shared/serverapi/workflow.go [hunk 2]
```diff
@@ -1000,13 +1015,143 @@ type WorkflowTaskMoveResponse struct {
...
+}
+
+type WorkflowRetainedPreviousWorktree struct {
+	Worktree WorkflowRegisteredWorktreeTopology `json:"worktree"`
+}
+
+type WorkflowRegisteredWorktreeTopology struct {
+	Variant    string                           `json:"variant"`
+	Registered *WorkflowRegisteredWorktreeFacts `json:"registered,omitempty"`
+}
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply shared/serverapi/workflow.go:1034]: No code change: this consolidation would conflict with the approved KENT-557 Design. Workflow remains JSON-only while Worktree is binary-Protobuf-only, and the User explicitly authorized this narrow Workflow-owned JSON seam: still-JSON Workflow fields use Workflow-owned DTOs/codecs mapped from independent generated Worktree values. Reusing the generated Worktree wire message here would make the unmigrated Workflow contract depend on the migrated Worktree binary contract and violate the hard-cutover ownership boundary.

File: api/proto/kent/api/worktree/worktree.proto [hunk 3]
```diff
@@ -662,29 +704,60 @@ message CreateTargetResolveResult {
...
-  option (buf.validate.message).cel = {
-    id: "kent.worktree.create_spec"
-    message: "create branch requires base_ref and branch_name; existing target requires base_ref only"
-    expression:
-      "(this.create_branch && has(this.base_ref) && has(this.branch_name)) || "
-      "(!this.create_branch && has(this.base_ref) && !has(this.branch_name))"
-  };
+  string session_id = 2 [(buf.validate.field).string.(kent.api.shared.nonblank) = true];
+  CreateSpec spec = 3 [(buf.validate.field).required = true];
+  optional string root_path = 4;
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply api/proto/kent/api/worktree/worktree.proto:741]: No code change: blank explicit `root_path` selecting automatic allocation is the approved existing behavior for KENT-557. The Design explicitly requires removing `min_len`, adding no nonblank constraint, and treating absent or trim-blank values as automatic allocation. Rejecting blank presence would be an out-of-scope product behavior change in this mechanical migration.
